### PR TITLE
support for numscript's set_tx_meta

### PR DESCRIPTION
**Pull request by Azorlogh**

### CODE REVIEW

File: pkg/api/controllers/script_controller.go [hunk 1]
```diff
@@ -13,7 +14,8 @@ import (
 
 type ScriptResponse struct {
 	ErrorResponse
-	Link string `json:"details,omitempty"`
+	Link string                           `json:"details,omitempty"`
+	Txs  []ledger.CommitTransactionResult `json:"txs,omitempty"`
```
Comment (gfyrag) [inline pkg/api/controllers/script_controller.go]: ledger.CommitTransactionResult is not supposed to be send on network without processing.
Errors inside the structure are internal and not well formatted in json.
In the best case, errors should be translated to api error codes. See https://github.com/numary/ledger/blob/a4fe5fa2e0507f71ec423277f9f7248a50973748/pkg/api/controllers/transaction_controller.go#L23

File: pkg/api/controllers/swagger.yaml [hunk 2]
```diff
@@ -456,6 +456,8 @@ components:
           type: string
         err:
           type: string
+        txs:
+          $ref: '#/components/schemas/CreateTransactionResponse'
```
Comment (gfyrag) [inline pkg/api/controllers/swagger.yaml]: CreateTransactionResponse represents the following structure : 
```
{
   "data": [ .... transactions ]
}
```
Seems not the good one for this case. The '#/components/schemas/TransactionCommitError" represent the good one (if error is properly translated, see previous comment).

File: pkg/ledger/executor_test.go [hunk 3]
```diff
@@ -82,6 +98,27 @@ func TestSend(t *testing.T) {
 	})
 }
 
+func TestNoVariables(t *testing.T) {
+	with(func(l *Ledger) {
+		var script core.Script
+		json.Unmarshal(
```
Comment (gfyrag) [inline pkg/ledger/executor_test.go:104]: I know, the error cannot occured, but if we can check ALL errors, i'm more happy :)

File: pkg/ledger/executor_test.go [hunk 4]
```diff
@@ -275,3 +343,52 @@ func TestMetadata(t *testing.T) {
 		assertBalance(t, l, "platform", "COIN", 15)
 	})
 }
+
+func TestSetTxMeta(t *testing.T) {
+	with(func(l *Ledger) {
+		defer l.Close(context.Background())
```
Comment (gfyrag) [inline pkg/ledger/executor_test.go:349]: Actually the 'with' method start an fx container which properly close the ledger at the end of the test, so it is useless to explicitely close it. But this is a no op and there is probably other tests with that.

**Review by gfyrag — COMMENTED** [review]
Some comments.
The main thing to modify is the swagger file and the ScriptResult response accordingly. 
We generate some code based on the openapi spec, and the swagger must be as clean as possible to be processed properly :)
The "txs" field can be json omitted to allow the PR to be merged and reintroduced later.

**Comment by Azorlogh** [comment]
Got it, I have reverted the txs field so it can be dealt with in a later PR

**Review by gfyrag — APPROVED** [review]

**Pull request merged**